### PR TITLE
Improve database download prompt when creating query

### PR DESCRIPTION
**Pull request by koesie10**
This improves the database download prompt when creating a query by removing it from the default flow. Instead, the user will get a notification with information on how they can download a database. There are 3 scenarios:

1. The user does not have a database for the language of the query:
2. 
Screenshot 2023-10-25 at 14 25 10

2. The user does have a database for the language of the query, but does not have it selected:

Screenshot 2023-10-25 at 14 26 04

3. The user does have a database for the language of the query and has it selected:

No notification

I'm very open to suggestions on how to better phrase these notifications, they are currently a bit wordy and repetitive.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Review by shati-patel — COMMENTED** [review]
Thanks for this! A couple of comments from me. I also agree that the notifications are a bit wordy, but I don't have much inspiration for other options 😅 

@jf205, do you have any opinions about the notification wording now that you can see them "in action"? 📝

File: extensions/ql-vscode/src/local-queries/skeleton-query-wizard.ts [hunk 1]
```diff
@@ -227,7 +242,24 @@ export class SkeletonQueryWizard {
...
-  private async downloadDatabase() {
+  private async promptDownloadDatabase() {
+    if (this.qlPackStoragePath === undefined) {
+      throw new Error("QL Pack storage path is undefined");
+    }
+
+    const openFileLink = this.openFileMarkdownLink;
+
+    const action = await showInformationMessageWithAction(
+      `New CodeQL query for ${this.language} ${openFileLink} created, but no CodeQL databases for ${this.language} were detected in your workspace. Would you like to download a CodeQL database for ${this.language} to analyze with ${openFileLink}?`,
```
[the remark this answers is not in the document]
Comment (koesie10) [reply extensions/ql-vscode/src/local-queries/skeleton-query-wizard.ts:253]: Thanks, I didn't know that function existed, but I've now used it for these notifications.

File: extensions/ql-vscode/src/local-queries/skeleton-query-wizard.ts [hunk 2]
```diff
@@ -280,14 +312,36 @@ export class SkeletonQueryWizard {
...
-      // select the found database
-      await this.databaseManager.setCurrentDatabaseItem(existingDatabaseItem);
+      const openFileLink = this.openFileMarkdownLink;
+
+      if (this.databaseManager.currentDatabaseItem !== existingDatabaseItem) {
+        // select the found database
+        await this.databaseManager.setCurrentDatabaseItem(existingDatabaseItem);
+
+        void window.showInformationMessage(
+          `New CodeQL query for ${this.language} ${openFileLink} created. We have automatically selected your existing CodeQL ${this.language} database ${existingDatabaseItem.name} for you to analyze with ${openFileLink}.`,
```
[the remark this answers is not in the document]
Comment (koesie10) [reply extensions/ql-vscode/src/local-queries/skeleton-query-wizard.ts:322]: I don't think so. There's no option to expand the notification when calling `showInformationMessage`, and the only reference to the expanded state I can find in the VS Code source code [will expand the notification by default if there are any actions](https://github.com/microsoft/vscode/blob/2b0bb4a46dc217fc9f4c8b6cb47c1a1e76e93426/src/vs/workbench/common/notifications.ts#L543).

File: extensions/ql-vscode/src/local-queries/skeleton-query-wizard.ts [hunk 3]
```diff
@@ -280,14 +312,36 @@ export class SkeletonQueryWizard {
...
+  private get openFileMarkdownLink() {
+    if (this.qlPackStoragePath === undefined) {
+      throw new Error("QL Pack storage path is undefined");
+    }
+
+    const openFileArgs = [
+      join(this.qlPackStoragePath, this.folderName, this.fileName),
+    ];
+    const queryString = encodeURI(JSON.stringify(openFileArgs));
+    return `[${this.fileName}](command:vscode.open?${queryString})`;
```
[the remark this answers is not in the document]
Comment (koesie10) [reply extensions/ql-vscode/src/local-queries/skeleton-query-wizard.ts]: I've tried using a `URI` to fix this. Let me know if it doesn't work, then we can pair on getting it to work on Windows.

[the remark this answers is not in the document]
Comment (shati-patel) [reply extensions/ql-vscode/src/local-queries/skeleton-query-wizard.ts]: Works for me now! Thanks ✨

**Review by shati-patel — APPROVED** [review]
Thanks for the edits! Looks good 🤩
